packages/server/src/api/controllers/row/alias.ts

```diff
@@ -0,0 +1,125 @@
+import { QueryJson, SearchFilters, Table, Row } from "@budibase/types"
+import { getDatasourceAndQuery } from "../../../sdk/app/rows/utils"
+import { cloneDeep } from "lodash"
+
+export default class AliasTables {
+  character: string
+  aliases: Record<string, string>
+  tableAliases: Record<string, string>
+  tableNames: string[]
+
+  constructor(tableNames: string[]) {
+    this.tableNames = tableNames
+    this.character = "a"
+    this.aliases = {}
+    this.tableAliases = {}
+  }
+
+  getAlias(tableName: string) {
+    if (this.aliases[tableName]) {
+      return this.aliases[tableName]
+    }
+    const char = this.character
+    this.aliases[tableName] = char
+    this.tableAliases[char] = tableName
+    this.character = String.fromCharCode(char.charCodeAt(0) + 1)
+    return char
+  }
+
+  aliasField(field: string) {
+    const tableNames = this.tableNames
+    if (field.includes(".")) {
+      const [tableName, column] = field.split(".")
+      if (tableNames.includes(tableName)) {
+        return `${this.getAlias(tableName)}.${column}`
```

Comment:
I'm also not sure this is strictly valid. Table names and column names can contain all sorts, and will likely need to be quoted to be compatiable with all of our users' tables.

Reply:
Quoting is handled deeper in the SQL layer, we split by `.` and handle quotation.